File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        lo = 0
        hi = len(nums) - 1
        
        ans = -1
        while lo <= hi:
            mid = int((lo + hi)/2)
            if nums[mid] == target:
                ans = mid
                break
            
            if nums[mid] < target:
                lo = mid + 1
            elif nums[mid] > target:
                hi = mid - 1
        
        return ans

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> ans(nums.size());
        int mul = 1;
        int cnt_zero = 0;
        for(int i = 0; i < nums.size() ; i++){
            if(nums[i] != 0){
                mul*= nums[i];
            }else{
                cnt_zero += 1;
            }
        }

        for(int i = 0; i < nums.size() ; i++){
            if(nums[i]!=0){
                ans[i] = (mul / nums[i]) * (cnt_zero == 0);
            }else{
                ans[i] = mul * (cnt_zero - 1 == 0);
            }
        }
        return ans;
    }
};
